backend/model/jobPoster.js

```diff
@@ -47,6 +48,8 @@ class JobPoster extends Job {
     return "poster";
   }
 
+  static preConfirmedJobs = [];
```

Comment:
Hey, yeh sure I'll add comments. But to just explain here as well what's happening.

The reason this bug was happening was because we were expecting the job posting transaction to finish within a single async function. And it turns out that the transaction wasn't being mined within the next 50 blocks, so this threw an error.

To get around this issue, I'm instead posting the job and then storing my values in this temporary array (preConfirmedJobs). And instead of waiting for a response from the post job request, I'm instead just listening for the postJobDescription event which is emitted once the new job is posted.

https://github.com/morphware/service/blob/4cbf23da52415bdec078f69deb69862e5ffcfe38/contracts/JobFactory.sol#L141

I then check if the event is related to the same job that I sent, if so, I move it from the 'preConfirmedJobs' array into the actual jobs array using the addToJump() method.

Reply:
Thanks! Super helpful